core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -353,7 +362,13 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
             HttpResponse serviceResponse;
             try {
                 req.init(reqCtx);
-                serviceResponse = service.serve(reqCtx, req);
+                serviceResponse = HttpResponse.from(() -> {
+                    try {
+                        return service.serve(reqCtx, req);
+                    } catch (Exception cause) {
+                        return HttpResponse.ofFailure(cause);
+                    }
+                }, serviceWorkerGroup);
```

Comment:
We need to use `serviceEventLoop`.

Reply:
Because the thread might be changed, the single thread model of this class is not worked anymore.
Could you check the data structure of the members of this class such as `unfinishedRequests` and fix it?
Also we need to change to use from `eventLoop` to `serviceEventLoop` in this method below.